### PR TITLE
added functionality in mmv1 to support configurable sweepers

### PR DESCRIPTION
**Pull request by shuyama1**
fixes https://github.com/hashicorp/terraform-provider-google/issues/16859

**Release Note Template for Downstream PRs (will be copied)**

```release-note:none

```

### CODE REVIEW

File: mmv1/provider/terraform/sweeper.rb [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+require 'uri'
+require 'api/object'
+require 'compile/core'
+require 'google/golang_utils'
+
+module Provider
+  class Terraform
+    # Support for schema ValidateFunc functionality.
+    class Sweeper < Google::YamlValidator
+      # The field checked by seeper to determine
```
Comment (c2thorn) [inline mmv1/provider/terraform/sweeper.rb:23]:
```suggestion
      # The field checked by sweeper to determine
```

File: mmv1/provider/terraform/sweeper.rb [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+require 'compile/core'
+require 'google/golang_utils'
+
+module Provider
+  class Terraform
+    # Support for schema ValidateFunc functionality.
+    class Sweeper < Google::YamlValidator
+      # The field checked by sweeper to determine
+      # eligibility for deletion for generated resources
+      attr_reader :tf_test_field
```
Comment (c2thorn) [inline mmv1/provider/terraform/sweeper.rb:25]: I was a little confused at what this was at first... can you think of a better name for this? I can't think of anything perfect, but maybe `sweepable_name_field` or something?

**Review by c2thorn — APPROVED** [review]
LGTM, just have an opinion on attribute name

**Comment by c2thorn** [comment]
As far as Go rewrite, we haven't gotten to sweepers yet, so this should be good to add right now.

File: mmv1/provider/terraform/sweeper.rb (hunk 2 above)
Comment (shuyama1) [reply mmv1/provider/terraform/sweeper.rb:25]: Yea, I was also struggled to find a better name for it. I'll update it with a more clear name, `sweepable_name_field` seems a good one

Comment (shuyama1) [reply mmv1/provider/terraform/sweeper.rb]: change it to `sweepable_identifier_field`. let me know if that makes sense

Comment (c2thorn) [reply mmv1/provider/terraform/sweeper.rb]: a lot clearer thanks!

**Review by c2thorn — APPROVED** [review]

**Pull request merged**